tests/github-glue.test.ts

```diff
@@ -175,23 +175,66 @@ test("pull requests", async () => {
+
+        const prCcInfo1 = await github.getPRInfo(owner, prData.number);
+        expect(prCcInfo1.body).toMatch(prCc);
+        const ccFound1 = prCcInfo1.body.match(new RegExp( prCc, "g"));
+        expect(ccFound1?.length).toEqual(1);
+
+        // Test with linefeeds present
+        const prBodyCc2 = `${prInfo.body}\r\n\r\nGlue`;
+        await github.updatePR(owner, prData.number, prBodyCc2);
+        await github.addPRCc(prData.html_url, prCc);
+        await github.addPRCc(prData.html_url, prCc);
+        await github.addPRCc(prData.html_url, prCc.toLowerCase());
         await github.addPRCc(prData.html_url, prCc2);
         await github.addPRCc(prData.html_url, prCcGitster);
         if (ghUser.email) {
             await github.addPRCc(prData.html_url, `PR Owner <${ghUser.email}>`);
+            await github.addPRCc(prData.html_url,
+                `PR Owner <${ghUser.email.toUpperCase()}>`);
         }
-        const prCcInfo = await github.getPRInfo(owner, prData.number);
-        expect(prCcInfo.body).toMatch(prCc);
-        const ccFound = prCcInfo.body.match(prCc);
-        expect(ccFound?.length).toEqual(1);
-        expect(prCcInfo.body).toMatch(prCc2);
-        expect(prCcInfo.body).not.toMatch(prCcGitster);
+        const prCcInfo2 = await github.getPRInfo(owner, prData.number);
+        expect(prCcInfo2.body).toMatch(prCc);
+        const ccFound2 = prCcInfo2.body.match(new RegExp( prCc, "g"));
+        expect(ccFound2?.length).toEqual(1);
+        expect(prCcInfo2.body).toMatch(prCc2);
+        expect(prCcInfo2.body).not.toMatch(prCcGitster);
         if (ghUser.email) {
-            expect(prCcInfo.body).not.toMatch(ghUser.email);
+            expect(prCcInfo2.body).not.
+                toMatch(new RegExp( ghUser.email, "i"));
         }
 
+        // Test with linefeeds and unknown footers
+        const prBodyCC3 = `${prInfo.body}\r\n \t\r\nbb: x\r\ncc: ${prCc}`;
+        await github.updatePR(owner, prData.number, prBodyCC3);
+        await github.addPRCc(prData.html_url, prCc);
+        await github.addPRCc(prData.html_url, prCc);
+        const prCcInfo3 = await github.getPRInfo(owner, prData.number);
+        expect(prCcInfo3.body).toMatch(prCc);
+        const ccFound3 = prCcInfo3.body.match(new RegExp( prCc, "g"));
+        expect(ccFound3?.length).toEqual(1);
+
+        // Test with linefeeds and unknown footer containing email
+        const prBodyCC4 = `${prInfo.body}\r\n \t\r\nbb: ${prCc}`;
+        await github.updatePR(owner, prData.number, prBodyCC4);
+        await github.addPRCc(prData.html_url, prCc);
+        await github.addPRCc(prData.html_url, prCc);
+        const prCcInfo4 = await github.getPRInfo(owner, prData.number);
+        const ccFound4 = prCcInfo4.body.match(new RegExp( prCc, "g"));
+        expect(ccFound4?.length).toEqual(2);
+
+        // Test to catch last block in body sort of a footer
+        const prBodyCC5 = `${prInfo.body}\r\n\r\nfoo\r\nbb: bar`;
```

Comment:
I might be missing something, but I don't think we look at ``prBodyN = `cc: ${prCc}`;`` nor at ``prBodyNPlus1 = `This is not a footer even if it contains\ncC: ${prCc}`;``.

Reply:
Adding more tests.  Separating the `cc` related tests, which do not need actual github updates, to use jest mock functions. They run much faster.

Reply:
Should be covered now.  Hopefully I have not missed more test cases.

The tests were redone to use mocks for the github calls since that is not the purpose of these tests.